### ISSUE TITLE
Sentry ASP.NET Core configuration issue with Serilog

### ISSUE THREAD

**Issue by KhnifesKyrillus** [getsentry/sentry-dotnet#2982]
### Package

Sentry

### .NET Flavor

.NET

### .NET Version

7.0.0

### OS

Windows

### SDK Version

3.41.3

### Steps to Reproduce

1. Setup Sentry with this config:
```json
"Sentry": {
    "AutoSessionTracking": true,
    "Dsn": "<DSN>",
    "SendDefaultPii": true,
    "MinimumBreadcrumbLevel": "Debug",
    "MinimumEventLevel": "Warning",
    "AttachStackTrace": true,
    "Debug": true,
    "DiagnosticLevel": "Error",
    "EnableTracing": true,
    "TracesSampleRate": 1.0,
    "Environment": "",
    "MaxRequestBodySize": "Always"
  },
```
2. Add this lines to your Program.cs:
```csharp
builder.WebHost.UseSentry();
```
3. Configure Serilog Sink:
```json
"WriteTo": [
      {
        "Name": "Sentry",
        "Args": {
          "MinimumBreadcrumbLevel": "Debug",
          "MinimumEventLevel": "Warning"
        }
      }
    ],
```

### Expected Result

The expected result is that the Sentry SDK is enabled and works.

### Actual Result

The actual result is that Sentry is not enabled until I add a Dsn to the Serilog Sink. As far as I understood the Sink should work without extra configuration if the ASP.NET Core integration ist configured

**Comment by KhnifesKyrillus** [comment]
The issue arises when the SDK is activated via the Sink; it becomes enabled prior to the `app.Run();`. On the other hand, when the SDK undergoes regular initialization, it kicks in only after the app.Run(); is triggered.

**Comment by jamescrosswell** [comment]
Hey @KhnifesKyrillus, thanks for reaching out. This is/was definitely confusing... there's a fairly detailed explanation of what's happening in:
- https://github.com/getsentry/sentry-dotnet/issues/2884

This has been addressed in [v4.0.0-beta3](https://github.com/getsentry/sentry-dotnet/blob/main/CHANGELOG.md#400-beta3) with [PR 2928](https://github.com/getsentry/sentry-dotnet/pull/2928) but if you'd rather stay one version 3.x of the API for a while, the TLDR; is that you need to configure `initializeSdk: false` in the Serilog Sink config (so that it doesn't try to "reinitialize" the SDK, with no DSN).
